bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -147,10 +147,14 @@ public function treeGetChildsByIdAction(Request $request, EventDispatcherInterfa
             $childsList->setCondition($condition);
             $childsList->setLimit($limit);
             $childsList->setOffset($offset);
-            $childsList->setOrderKey(
-                sprintf('CAST(objects.o_%s AS CHAR CHARACTER SET utf8) COLLATE utf8_general_ci ASC', $object->getChildrenSortBy()),
-                false
-            );
+            if($object->getChildrenSortBy() === 'index') {
+                $childsList->setOrderKey('objects.o_index ASC', false);
+            } else {
+                $childsList->setOrderKey(
+                    sprintf('CAST(objects.o_%s AS CHAR CHARACTER SET utf8) COLLATE utf8_general_ci ASC', $object->getChildrenSortBy()),
```

Comment:
+1 This is nice. But maybe consider testing if [utf8mb4_unicode_ci](https://stackoverflow.com/a/766996/4919561) works?

Reply:
Why? Casting always needs more resources than sorting by the original (indexed!) column.